model.py

```diff
@@ -8795,9 +8812,51 @@ class ExternalIntegration(Base):
     to a third-party API.
     """
 
+    # Utilities
+    CDN = Configuration.CDN_INTEGRATION
+    ELASTICSEARCH = Configuration.ELASTICSEARCH_INTEGRATION
+    AMAZON_S3 = Configuration.S3_INTEGRATION
+
+    # Simplified
+    CIRCULATION_MANAGER = Configuration.CIRCULATION_MANAGER_INTEGRATION
+    CONTENT_SERVER = Configuration.CONTENT_SERVER_INTEGRATION
+    METADATA_WRANGLER = Configuration.METADATA_WRANGLER_INTEGRATION
+    PATRON_WEB_CLIENT = u'Patron Web Client'
+
+    # Metadata
+    BIBBLIO = u'Bibblio'
+    CONTENT_CAFE = u'Content Cafe'
+    NOVELIST = Configuration.NOVELIST_INTEGRATION
+    NYPL_SHADOWCAT = u'Shadowcat'
+    NYT = Configuration.NYT_INTEGRATION
+    STAFF_PICKS = u'Staff Picks'
+
+    # Analytics
+    GOOGLE_ANALYTICS = u'Google Analytics'
+
+    # Patron Authentication
+    ADOBE_VENDOR_ID = u'Adobe Vendor ID'
+
+    # Admin Authentication
+    ADMIN_AUTH_TYPE = u'admin_auth'
+    GOOGLE_OAUTH = u'Google OAuth'
+    ADMIN_AUTH_PROVIDERS = [GOOGLE_OAUTH]
+
+    PROVIDERS = [
+        ADOBE_VENDOR_ID, AMAZON_S3, BIBBLIO, CDN, CIRCULATION_MANAGER,
+        CONTENT_CAFE, CONTENT_SERVER, ELASTICSEARCH, GOOGLE_ANALYTICS,
+        GOOGLE_OAUTH, METADATA_WRANGLER, NOVELIST, NYPL_SHADOWCAT,
+        PATRON_WEB_CLIENT, NYT, STAFF_PICKS,
+    ]
+
     __tablename__ = 'externalintegrations'
     id = Column(Integer, primary_key=True)
 
+    # If this integration isn't related to a Collection, it should
+    # have a provider and type.
+    provider = Column(Unicode, nullable=True)
```

Comment:
Monitors and CoverageProviders would definitely need to take a list of accepted PROTOCOLS instead of a single covered PROTOCOL or `None` in that case.